src/ast/ast_validations.c

```diff
@@ -680,6 +681,17 @@ static AST_Validation _Validate_WITH_Clauses(const AST *ast, char **reason) {
 	const cypher_astnode_t *with_clause = AST_GetClause(ast, CYPHER_AST_WITH);
 	if(with_clause == NULL) return AST_VALID;
 
+	// Verify that each WITH projection either is aliased or is itself an identifier.
+	uint projection_count = cypher_ast_with_nprojections(with_clause);
+	for(uint i = 0; i < projection_count; i ++) {
+		const cypher_astnode_t *proj = cypher_ast_with_get_projection(with_clause, i);
+		if(!cypher_ast_projection_get_alias(proj) &&
+		   cypher_astnode_type(cypher_ast_projection_get_expression(proj)) != CYPHER_AST_IDENTIFIER) {
```

Comment:
`cypher_astnode_type(cypher_ast_projection_get_expression(proj)) != CYPHER_AST_IDENTIFIER`
Is this suggesting that a WITH projection must be an identifiers?
In the case `WITH MAX(n.v) AS maximum` does  `MAX(n.v)` considered as an identifier, or am I missing something?

Reply:
All WITH projections must either be aliased or already be simple identifiers. We check for aliases first with the first condition, `!cypher_ast_projection_get_alias(proj)`, and check for the latter with the second condition.

Do you think a nested `if` that separated the conditions would be better for clarity? I wrote it both ways, but couldn't decide.